src/shims/posix/fs.rs

```diff
@@ -78,12 +76,11 @@ impl FileDescriptor for FileHandle {
         }
     }
 
-    fn dup<'tcx>(&mut self) -> InterpResult<'tcx, io::Result<Box<dyn FileDescriptor>>> {
-        self
-            .as_file_handle()
-            .map(|FileHandle { file, writable }|
-                 file.try_clone().map(|duplicated|
-                                      Box::new(FileHandle { file: duplicated, writable: *writable }) as Box<dyn FileDescriptor>))
+    fn dup<'tcx>(&mut self) -> io::Result<Box<dyn FileDescriptor>> {
+        self.file
+            .try_clone()
+            .map(|duplicated|
+                 Box::new(FileHandle { file: duplicated, writable: self.writable }) as Box<dyn FileDescriptor>)
```

Comment:
Is the `as Box<...>` needed? I think type inference should be able to figure this out

Reply:
It's not able to, for some reason:

```
   = note: expected enum `std::result::Result<std::boxed::Box<(dyn FileDescriptor + 'static)>, _>`
              found enum `std::result::Result<std::boxed::Box<FileHandle>, _>`
```

Reply:
Strange.
Maybe if you avoid `map`?
```rust
let duplicated = self.file.try_clone()?;
Ok(Box::new(FileHandle { ... }))
```

Reply:
That works. Changed this.